pittapi/news.py

```diff
@@ -17,56 +17,109 @@
 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 """
 
-import math
-import requests
-import grequests
-from typing import Any
-
-sess = requests.session()
-
-
-def _load_n_items(feed: str, max_news_items: int):
-    payload = {
-        "feed": feed,
-        "id": "",
-        "_object": "kgoui_Rcontent_I0_Rcontent_I0",
-        "start": 0,
-    }
-
-    request_objs = []
-    for i in range(int(math.ceil(max_news_items / 10))):
-        payload["start"] = i * 10
-        request_objs.append(grequests.get("https://m.pitt.edu/news/index.json", params=payload))
-
-    responses = grequests.imap(request_objs)
-
-    return responses
+from __future__ import annotations
 
-
-def get_news(feed: str = "main_news", max_news_items: int = 10) -> list[dict[str, Any]]:
-    # feed indicates the desired news feed
-    # 'main_news'      - main news
-    # 'cssd'           - student announcements, on my pitt
-    # 'news_chronicle' - the Pitt Chronicle news
-    # 'news_alerts'    - crime alerts
-
-    news = []
-
-    resps = _load_n_items(feed, max_news_items)
-    resps = [r.json()["response"]["regions"][0]["contents"] for r in resps]
-
-    for resp in resps:
-        for data in resp:
-            fields = data["fields"]
-            if fields["type"] == "loadMore":
-                continue
-
-            # TODO: Look into why this gives a Type Error during the news alert test.
-            try:
-                title = fields["title"]
-                url = "https://m.pitt.edu" + fields["url"]["formatted"]
-                news.append({"title": title, "url": url})
-            except TypeError:
-                continue
-
-    return news[:max_news_items]
+import math
+from requests_html import Element, HTMLResponse, HTMLSession
+from typing import Literal, NamedTuple
+
+NUM_ARTICLES_PER_PAGE = 20
```

Comment:
Since this is really only used as the default value for `max_num_results` in `get_articles_by_topic`, does this need to be a global variable? What are the pros of this as opposed to `max_num_results: int = 20`?

Reply:
I see how you use it in tests later. Sensible. Feel free to ignore this entirely

Reply:
I also use the value to calculate how many pages of results are required:
```py
num_pages = math.ceil(max_num_results / NUM_ARTICLES_PER_PAGE)
```
so I just made the value a global "constant" to avoid repeating it in multiple places.

Reply:
That's a great call. Ignore my comment entirely, I missed that usage!